### ISSUE TITLE
Add Zenscript Support?

### ISSUE THREAD

**Issue by kriNon** [DoubleDoorDevelopment/ItemBlacklist#26]
Hey,
Was thinking that it'd be awesome to be able to blacklist items through zenscript, it'd make it really convenient to be able to ban items and could be very useful for packdevs.

Thanks!

**Comment by Claycorp** [comment]
Not entirely sure what this would bring that the mod already does on it's own?

**Comment by kriNon** [comment]
Well mostly I think that it would be awesome to be able to remove recipes for items, hide from NEI, remove an item from loottables and add to ItemBlacklist in the same place. It would also make it possible to make a zenscript function that would do all of the above for an item at once.

Mostly it'd just make it easier to remove an item from the game, without having to try to remove it in a bunch of different places.

**Comment by Claycorp** [comment]
Doesn't JEI handle all the item hiding on it's own? Last time I used CraftTweaker in 1.10 I dont recall it having hide options. 

Also I dont think you quite understand the purpose of this mod. It's to block people from using certain stuff in certain dims or to block items that cant be removed in any other way. If you cant obtain it, adding it to the blacklist does nothing.

**Comment by kriNon** [comment]
Yeah JEI does handle item hiding, however due to JEI-zenscript compatability it's possible to also hide items using zenscript.

I believe I do understand the purpose of the mod.  My thoughts are that this mod would be useful because it would prevent any instances of people finding alternate ways of getting items, for example a mod may add a way of getting an item that can't be disabled otherwise.

If you really don't want to do it, I'm not trying to push you to do it, I just think it would be useful and appreciated.

**Comment by Claycorp** [comment]
I get you aren't trying to push for it, I'm just trying to find the value in trying to add it. I don't have as much time as I did before and I mainly port the older mods forward these days as most of them where made back in 1.7.10 and might not be that useful/the best way to do it anymore. Also they where written by Dries who has a much better understanding of many aspects compared to me. Thus it will take me more time to work through the code and how it works totally.

I just currently think the way it works can't be any easier. Aside for allowing mod wide bans and adding a ban all items held in the main inventory mode for 1.13. (See #25 for that) Perhaps if the mod ever needs to be totally remade I will look into ZenScript support but as it is currently I don't see the need, Sorry.

I will however leave this issue open if other people wish to add it and want to submit a PR they are more than welcome to. Also to remind me of the idea in case of a rewrite.

**Comment by Fireztonez** [comment]
Sincerely, adding crafttweaker support will add a lot of possibilities questions to blacklist items really fast and easy.

And yes JEI have crafttweaker support now and there have hide command and now removeAndHide questions to remove recipe at the same time...

But, other thing, with crafttweaker you can use loops, questions of disable multiple items really easy with a simple array variable or associative array questions to do with dimensions list...

Crafttweaker is an awesome tool for modpack dev and dev docs is awesome aswell:

https://crafttweaker.readthedocs.io/en/latest/
